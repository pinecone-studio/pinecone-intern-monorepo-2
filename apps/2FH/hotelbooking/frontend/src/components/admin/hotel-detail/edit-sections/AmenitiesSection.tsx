'use client';
import { Label } from '@/components/ui/Label';
import { Checkbox } from '@/components/ui/Checkbox';

interface AmenitiesSectionProps {
  formData: any;
  handleInputChange: (_field: string, _value: any) => void;
}

export const AmenitiesSection = ({ formData, handleInputChange }: AmenitiesSectionProps) => {
  const availableAmenities = [
    'POOL',
    'GYM',
    'RESTAURANT',
    'BAR',
    'WIFI',
    'PARKING',
    'FITNESS_CENTER',
    'BUSINESS_CENTER',
    'MEETING_ROOMS',
    'CONFERENCE_ROOMS',
    'ROOM_SERVICE',
    'AIR_CONDITIONING',
    'AIRPORT_TRANSFER',
    'FREE_WIFI',
    'FREE_PARKING',
    'FREE_CANCELLATION',
    'SPA',
    'PETS_ALLOWED',
    'SMOKING_ALLOWED',
    'LAUNDRY_FACILITIES',
  ];

  return (
    <div className="space-y-4">
      <Label>Amenities</Label>
      <div className="grid grid-cols-2 gap-3 max-h-60 overflow-y-auto">
        {availableAmenities.map((amenity) => (
          <div key={amenity} className="flex items-center space-x-2">
            <Checkbox
              id={amenity}
              checked={formData.amenities.includes(amenity)}
              onCheckedChange={(checked) => {
                if (checked) {
                  handleInputChange('amenities', [...formData.amenities, amenity]);
                } else {
                  handleInputChange(
                    'amenities',
                    formData.amenities.filter((a: string) => a !== amenity)
                  );
                }
              }}
            />
            <Label htmlFor={amenity} className="text-sm">
              {amenity.replace(/_/g, ' ')}
            </Label>
          </div>
        ))}
      </div>
    </div>
  );
};
