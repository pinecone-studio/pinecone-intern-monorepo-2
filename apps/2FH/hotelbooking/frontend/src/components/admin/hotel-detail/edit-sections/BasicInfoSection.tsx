'use client';
import { Input } from '@/components/ui/Input';
import { Textarea } from '@/components/ui/Textarea';
import { Label } from '@/components/ui/Label';

interface BasicInfoSectionProps {
  formData: any;
  handleInputChange: (_field: string, _value: any) => void;
}

export const BasicInfoSection = ({ formData, handleInputChange }: BasicInfoSectionProps) => {
  return (
    <div className="space-y-4">
      <div>
        <Label htmlFor="name" className="text-sm font-medium text-gray-700">
          Hotel Name
        </Label>
        <Input id="name" value={formData.name} onChange={(e) => handleInputChange('name', e.target.value)} />
      </div>
      <div>
        <Label htmlFor="description" className="text-sm font-medium text-gray-700">
          Description
        </Label>
        <Textarea id="description" value={formData.description} onChange={(e) => handleInputChange('description', e.target.value)} rows={4} />
      </div>
      <div className="grid grid-cols-2 gap-4">
        <div>
          <Label htmlFor="phone" className="text-sm font-medium text-gray-700">
            Phone Number
          </Label>
          <Input id="phone" value={formData.phone} onChange={(e) => handleInputChange('phone', e.target.value)} />
        </div>
        <div>
          <Label htmlFor="stars" className="text-sm font-medium text-gray-700">
            Stars (Read-only)
          </Label>
          <Input id="stars" type="number" min="1" max="5" value={formData.stars} disabled className="bg-gray-100" />
        </div>
      </div>
      <div className="grid grid-cols-2 gap-4">
        <div>
          <Label htmlFor="country" className="text-sm font-medium text-gray-700">
            Country
          </Label>
          <Input id="country" value={formData.country} onChange={(e) => handleInputChange('country', e.target.value)} placeholder="Enter country..." />
        </div>
        <div>
          <Label htmlFor="city" className="text-sm font-medium text-gray-700">
            City
          </Label>
          <Input id="city" value={formData.city} onChange={(e) => handleInputChange('city', e.target.value)} placeholder="Enter city name..." />
        </div>
      </div>
      <div>
        <Label htmlFor="location" className="text-sm font-medium text-gray-700">
          Location
        </Label>
        <Input id="location" value={formData.location} onChange={(e) => handleInputChange('location', e.target.value)} />
      </div>
      <div>
        <Label htmlFor="rating" className="text-sm font-medium text-gray-700">
          User Rating (Read-only)
        </Label>
        <Input id="rating" type="number" step="0.1" min="0" max="10" value={formData.rating} disabled className="bg-gray-100" />
      </div>
    </div>
  );
};
