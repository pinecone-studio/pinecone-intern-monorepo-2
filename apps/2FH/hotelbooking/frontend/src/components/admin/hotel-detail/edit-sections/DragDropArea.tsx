import React from 'react';
import { Button } from '@/components/ui/button';

interface DragDropAreaProps {
  dragActive: boolean;
  uploading: boolean;
  fileInputRef: React.RefObject<HTMLInputElement>;
  handleDragEnter: (_e: React.DragEvent) => void;
  handleDragLeave: (_e: React.DragEvent) => void;
  handleDragOver: (_e: React.DragEvent) => void;
  handleDrop: (_e: React.DragEvent) => void;
  handleFileInput: (_e: React.ChangeEvent<HTMLInputElement>) => void;
}

export const DragDropArea = ({ dragActive, uploading, fileInputRef, handleDragEnter, handleDragLeave, handleDragOver, handleDrop, handleFileInput }: DragDropAreaProps) => (
  <div
    className={`border-2 border-dashed rounded-lg p-8 text-center transition-colors ${dragActive ? 'border-blue-500 bg-blue-50' : 'border-gray-300 hover:border-gray-400'}`}
    onDragEnter={handleDragEnter}
    onDragLeave={handleDragLeave}
    onDragOver={handleDragOver}
    onDrop={handleDrop}
  >
    <div className="space-y-4">
      <div className="mx-auto w-16 h-16 bg-gray-100 rounded-full flex items-center justify-center">
        <svg className="w-8 h-8 text-gray-400" fill="none" stroke="currentColor" viewBox="0 0 24 24">
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12" />
        </svg>
      </div>

      <div>
        <p className="text-lg font-medium text-gray-900">{uploading ? 'Uploading...' : 'Drop images here or click to upload'}</p>
        <p className="text-sm text-gray-500 mt-1">Supports JPG, PNG, GIF up to 10MB each</p>
      </div>

      <Button type="button" variant="outline" onClick={() => fileInputRef.current?.click()} disabled={uploading} className="mt-4">
        {uploading ? 'Uploading...' : 'Choose Files'}
      </Button>

      <input ref={fileInputRef} type="file" multiple accept="image/*" onChange={handleFileInput} className="hidden" />
    </div>
  </div>
);
