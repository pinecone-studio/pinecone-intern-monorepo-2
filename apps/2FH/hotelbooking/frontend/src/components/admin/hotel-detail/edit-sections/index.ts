export { BasicInfoSection } from './BasicInfoSection';
export { LocationSection } from './LocationSection';
export { AmenitiesSection } from './AmenitiesSection';
export { LanguagesSection } from './LanguagesSection';
export { PoliciesSection } from './PoliciesSection';
export { FAQSection } from './FaqSection';
export { DetailsSection } from './DetailsSection';
export { ImagesSection } from './ImagesSection';
