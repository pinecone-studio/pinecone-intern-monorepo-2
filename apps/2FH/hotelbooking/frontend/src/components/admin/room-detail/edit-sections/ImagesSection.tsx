/* eslint-disable  */
'use client';
import { useState, useRef } from 'react';
import { Button } from '@/components/ui/button';
import { Label } from '@/components/ui/label';
import { Input } from '@/components/ui/input';
import { X, Upload, Image as ImageIcon } from 'lucide-react';

interface ImagesSectionProps {
  room: any;
  handleInputChange: (field: string, value: any) => void;
}

export const ImagesSection = ({ room, handleInputChange }: ImagesSectionProps) => {
  const [previewImages, setPreviewImages] = useState<string[]>(
    Array.isArray(room.imageURL) ? room.imageURL.filter((img: string) => img !== null && img !== undefined && img !== '') : room.imageURL ? [room.imageURL] : []
  );
  const fileInputRef = useRef<HTMLInputElement>(null);

  const handleFileUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
    const files = event.target.files;
    if (files) {
      const newImages: string[] = [];

      Array.from(files).forEach((file) => {
        if (file.type.startsWith('image/')) {
          const reader = new FileReader();
          reader.onload = (e) => {
            const result = e.target?.result as string;
            newImages.push(result);

            if (newImages.length === files.length) {
              const updatedImages = [...previewImages, ...newImages];
              setPreviewImages(updatedImages);
              handleInputChange('imageURL', updatedImages);
            }
          };
          reader.readAsDataURL(file);
        }
      });
    }
  };

  const removeImage = (index: number) => {
    const updatedImages = previewImages.filter((_, i) => i !== index);
    setPreviewImages(updatedImages);
    handleInputChange('imageURL', updatedImages);
  };

  const handleUploadClick = () => {
    fileInputRef.current?.click();
  };

  return (
    <div className="space-y-6">
      <div>
        <Label>Room Images</Label>
        <div className="mt-2">
          <Button
            type="button"
            variant="outline"
            onClick={handleUploadClick}
            className="w-full h-32 border-2 border-dashed border-gray-300 hover:border-gray-400 flex flex-col items-center justify-center gap-2"
          >
            <Upload size={24} className="text-gray-400" />
            <span className="text-gray-600">Click to upload images</span>
            <span className="text-sm text-gray-400">PNG, JPG, GIF up to 10MB each</span>
          </Button>
          <Input ref={fileInputRef} type="file" multiple accept="image/*" onChange={handleFileUpload} className="hidden" />
        </div>
      </div>

      {previewImages.length > 0 && (
        <div>
          <Label>Image Previews ({previewImages.length})</Label>
          <div className="mt-2 grid grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4">
            {previewImages.map((image, index) => (
              <div key={index} className="relative group">
                <div className="aspect-square rounded-lg overflow-hidden border border-gray-200">
                  <img
                    src={image}
                    alt={`Room image ${index + 1}`}
                    className="w-full h-full object-cover"
                    onError={(e) => {
                      const target = e.target as HTMLImageElement;
                      target.style.display = 'none';
                    }}
                  />
                </div>
                <Button
                  type="button"
                  variant="destructive"
                  size="sm"
                  className="absolute -top-2 -right-2 h-6 w-6 rounded-full p-0 opacity-0 group-hover:opacity-100 transition-opacity"
                  onClick={() => removeImage(index)}
                >
                  <X size={12} />
                </Button>
                <div className="absolute bottom-0 left-0 right-0 bg-black bg-opacity-50 text-white text-xs p-1 opacity-0 group-hover:opacity-100 transition-opacity">Image {index + 1}</div>
              </div>
            ))}
          </div>
        </div>
      )}

      {previewImages.length === 0 && (
        <div className="text-center py-8 text-gray-500">
          <ImageIcon size={48} className="mx-auto mb-2 text-gray-300" />
          <p>No images uploaded yet</p>
          <p className="text-sm">Upload images to preview them here</p>
        </div>
      )}
    </div>
  );
};
