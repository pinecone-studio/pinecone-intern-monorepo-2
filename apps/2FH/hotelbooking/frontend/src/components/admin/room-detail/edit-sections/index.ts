export { BasicInfoSection } from './BasicInfoSection';
export { AmenitiesSection } from './AmenitiesSection';
export { ImagesSection } from './ImagesSection';
export { DetailsSection } from './DetailsSection';


