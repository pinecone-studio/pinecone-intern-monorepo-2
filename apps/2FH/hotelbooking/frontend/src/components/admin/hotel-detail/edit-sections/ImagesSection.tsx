'use client';
import { useState, useRef } from 'react';
import { Button } from '@/components/ui/Button';
import { Label } from '@/components/ui/Label';
import { DragDropArea } from './DragDropArea';
import Image from 'next/image';
interface ImagesSectionProps {
  formData: any;
  handleInputChange: (_field: string, _value: any) => void;
}
const ImageItem = ({ image, onRemove }: { image: string; onRemove: () => void }) => (
  <div className="relative group">
    <div className="relative aspect-square rounded-lg overflow-hidden border">
      <Image src={image} alt={`Hotel image`} fill className="object-cover" />
      <div className="absolute inset-0 bg-black bg-opacity-0 group-hover:bg-opacity-30 transition-all duration-200 flex items-center justify-center">
        <Button size="sm" variant="destructive" onClick={onRemove} className="opacity-0 group-hover:opacity-100 transition-opacity duration-200 bg-red-600 hover:bg-red-700">
          Remove
        </Button>
      </div>
    </div>
    <p className="text-xs text-gray-500 mt-1 text-center">Image</p>
  </div>
);
const useImageUpload = (formData: any, handleInputChange: (_field: string, _value: any) => void) => {
  const [dragActive, setDragActive] = useState(false);
  const [uploading, setUploading] = useState(false);
  const fileInputRef = useRef<HTMLInputElement>(null);
  const preventDefault = (e: React.DragEvent) => {
    e.preventDefault();
    e.stopPropagation();
  };
  const uploadToCloudinary = async (file: File): Promise<string> => {
    const cloudName = 'dxjdxefkk';
    const uploadPreset = 'RoomImage';
    const formData = new FormData();
    formData.append('file', file);
    formData.append('upload_preset', uploadPreset);
    const response = await fetch(`https://api.cloudinary.com/v1_1/${cloudName}/image/upload`, {
      method: 'POST',
      body: formData,
    });
    if (!response.ok) {
      const errorData = await response.json().catch(() => ({}));
      throw new Error(errorData.error?.message || 'Failed to upload image to Cloudinary');
    }
    const data = await response.json();
    return data.secure_url;
  };
  const processImageFile = async (file: File): Promise<string> => {
    try {
      return await uploadToCloudinary(file);
    } catch (error) {
      console.error('Error uploading to Cloudinary:', error);
      // Fallback to local file reading for test environment
      if (process.env.NODE_ENV === 'test') {
        return new Promise((resolve, reject) => {
          const reader = new FileReader();
          reader.onload = (e) => resolve(e.target?.result as string);
          reader.onerror = reject;
          reader.readAsDataURL(file);
        });
      }
      throw error;
    }
  };

  const handleFiles = async (files: FileList) => {
    setUploading(true);
    try {
      const imageFiles = Array.from(files).filter((file) => file.type.startsWith('image/'));
      const newImages = await Promise.all(imageFiles.map(processImageFile));
      handleInputChange('images', [...formData.images, ...newImages]);
    } catch (error) {
      console.error('Error uploading images:', error);
    } finally {
      setUploading(false);
    }
  };

  const handleDragEnter = (e: React.DragEvent) => {
    preventDefault(e);
    setDragActive(true);
  };
  const handleDragLeave = (e: React.DragEvent) => {
    preventDefault(e);
    setDragActive(false);
  };
  const handleDragOver = (e: React.DragEvent) => {
    preventDefault(e);
    setDragActive(true);
  };
  const handleDrop = (e: React.DragEvent) => {
    preventDefault(e);
    setDragActive(false);
    if (e.dataTransfer.files?.[0]) {
      handleFiles(e.dataTransfer.files);
    }
  };
  const handleFileInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (e.target.files?.[0]) {
      handleFiles(e.target.files);
    }
  };
  return {
    dragActive,
    uploading,
    fileInputRef,
    handleDragEnter,
    handleDragLeave,
    handleDragOver,
    handleDrop,
    handleFileInput,
  };
};
export const ImagesSection = ({ formData, handleInputChange }: ImagesSectionProps) => {
  const { dragActive, uploading, fileInputRef, handleDragEnter, handleDragLeave, handleDragOver, handleDrop, handleFileInput } = useImageUpload(formData, handleInputChange);
  const removeImage = (index: number) => {
    const newImages = formData.images.filter((_: string, i: number) => i !== index);
    handleInputChange('images', newImages);
  };
  return (
    <div className="space-y-6">
      <div>
        <Label className="text-lg font-medium">Hotel Images</Label>
        <p className="text-sm text-gray-600 mt-1">Upload images from your computer or phone</p>
      </div>
      {formData.images.length > 0 && (
        <div>
          <h4 className="font-medium text-gray-900 mb-3">Current Images</h4>
          <div className="grid grid-cols-2 md:grid-cols-3 gap-4">
            {formData.images.map((image: string, index: number) => (
              <ImageItem key={index} image={image} onRemove={() => removeImage(index)} />
            ))}
          </div>
        </div>
      )}
      <DragDropArea
        dragActive={dragActive}
        uploading={uploading}
        fileInputRef={fileInputRef as React.RefObject<HTMLInputElement>}
        handleDragEnter={handleDragEnter}
        handleDragLeave={handleDragLeave}
        handleDragOver={handleDragOver}
        handleDrop={handleDrop}
        handleFileInput={handleFileInput}
      />
      {uploading && (
        <div className="text-center py-4">
          <div className="inline-flex items-center gap-2">
            <div className="animate-spin rounded-full h-4 w-4 border-b-2 border-blue-600"></div>
            <span className="text-sm text-gray-600">Processing images...</span>
          </div>
        </div>
      )}
    </div>
  );
};