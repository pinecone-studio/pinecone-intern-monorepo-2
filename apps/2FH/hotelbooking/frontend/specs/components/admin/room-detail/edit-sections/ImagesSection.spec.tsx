/* eslint-disable */
import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { ImagesSection } from '@/components/admin/room-detail/edit-sections/ImagesSection';

// Mock FileReader
const mockFileReader = {
  readAsDataURL: jest.fn(),
  onload: null as any,
  result: null as any,
};

global.FileReader = jest.fn(() => mockFileReader) as any;

describe('ImagesSection', () => {
  const mockRoom = {
    imageURL: ['https://example.com/image1.jpg', 'https://example.com/image2.jpg'],
  };

  const defaultProps = {
    room: mockRoom,
    handleInputChange: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
    mockFileReader.onload = null;
    mockFileReader.result = null;
  });

  it('renders upload section correctly', () => {
    render(<ImagesSection {...defaultProps} />);

    expect(screen.getByText('Room Images')).toBeInTheDocument();
    expect(screen.getByText('Click to upload images')).toBeInTheDocument();
    expect(screen.getByText('PNG, JPG, GIF up to 10MB each')).toBeInTheDocument();
  });

  it('displays existing images', () => {
    render(<ImagesSection {...defaultProps} />);

    expect(screen.getByText('Image Previews (2)')).toBeInTheDocument();
    expect(screen.getByAltText('Room image 1')).toBeInTheDocument();
    expect(screen.getByAltText('Room image 2')).toBeInTheDocument();
  });

  it('handles room with single image as string', () => {
    const roomWithStringImage = {
      imageURL: 'https://example.com/single-image.jpg',
    };

    render(<ImagesSection {...defaultProps} room={roomWithStringImage} />);

    expect(screen.getByText('Image Previews (1)')).toBeInTheDocument();
    expect(screen.getByAltText('Room image 1')).toBeInTheDocument();
  });

  it('handles room with no images', () => {
    const roomWithNoImages = {
      imageURL: [],
    };

    render(<ImagesSection {...defaultProps} room={roomWithNoImages} />);

    expect(screen.getByText('No images uploaded yet')).toBeInTheDocument();
    expect(screen.getByText('Upload images to preview them here')).toBeInTheDocument();
  });

  it('handles room with null imageURL', () => {
    const roomWithNullImages = {
      imageURL: null,
    };

    render(<ImagesSection {...defaultProps} room={roomWithNullImages} />);

    expect(screen.getByText('No images uploaded yet')).toBeInTheDocument();
  });

  it('handles room with undefined imageURL', () => {
    const roomWithUndefinedImages = {
      imageURL: undefined,
    };

    render(<ImagesSection {...defaultProps} room={roomWithUndefinedImages} />);

    expect(screen.getByText('No images uploaded yet')).toBeInTheDocument();
  });

  it('handles file upload click', () => {
    render(<ImagesSection {...defaultProps} />);

    const uploadButton = screen.getByText('Click to upload images');
    fireEvent.click(uploadButton);

    // File input should be triggered (we can't assert directly)
    expect(uploadButton).toBeInTheDocument();
  });

  it('does nothing when files is null', () => {
    render(<ImagesSection {...defaultProps} />);

    const fileInput = document.querySelector('input[type="file"]') as HTMLInputElement;

    // Simulate change event with null files
    fireEvent.change(fileInput, { target: { files: null } });

    // Should not call handleInputChange
    expect(defaultProps.handleInputChange).not.toHaveBeenCalled();
  });

  it('handles file selection with valid images', async () => {
    const mockFile = new File(['image content'], 'test.jpg', { type: 'image/jpeg' });
    const mockFileList = {
      0: mockFile,
      length: 1,
      item: (index: number) => (index === 0 ? mockFile : null),
    };

    render(<ImagesSection {...defaultProps} />);

    const fileInput = screen.getByRole('button', { name: /click to upload images/i }).parentElement?.querySelector('input[type="file"]') as HTMLInputElement;

    if (fileInput) {
      Object.defineProperty(fileInput, 'files', {
        value: mockFileList,
        writable: false,
      });

      mockFileReader.result = 'data:image/jpeg;base64,test-image-data';

      fireEvent.change(fileInput);

      if (mockFileReader.onload) {
        mockFileReader.onload({ target: { result: 'data:image/jpeg;base64,test-image-data' } } as any);
      }

      await waitFor(() => {
        expect(defaultProps.handleInputChange).toHaveBeenCalled();
      });
    }
  });

  it('handles file selection with non-image files', () => {
    const mockFile = new File(['text content'], 'test.txt', { type: 'text/plain' });
    const mockFileList = {
      0: mockFile,
      length: 1,
      item: (index: number) => (index === 0 ? mockFile : null),
    };

    render(<ImagesSection {...defaultProps} />);

    const fileInput = document.querySelector('input[type="file"]') as HTMLInputElement;
    Object.defineProperty(fileInput, 'files', {
      value: mockFileList,
      writable: false,
    });

    fireEvent.change(fileInput);

    expect(defaultProps.handleInputChange).not.toHaveBeenCalled();
  });

  it('handles file processing with multiple images', async () => {
    const mockFile1 = new File(['image1'], 'test1.jpg', { type: 'image/jpeg' });
    const mockFile2 = new File(['image2'], 'test2.png', { type: 'image/png' });
    const mockFileList = {
      0: mockFile1,
      1: mockFile2,
      length: 2,
      item: (index: number) => (index < 2 ? [mockFile1, mockFile2][index] : null),
    };

    render(<ImagesSection {...defaultProps} />);

    const fileInput = screen.getByRole('button', { name: /click to upload images/i }).parentElement?.querySelector('input[type="file"]') as HTMLInputElement;

    if (fileInput) {
      Object.defineProperty(fileInput, 'files', {
        value: mockFileList,
        writable: false,
      });

      mockFileReader.result = 'data:image/jpeg;base64,test-image-data-1';
      fireEvent.change(fileInput);

      if (mockFileReader.onload) {
        mockFileReader.onload({ target: { result: 'data:image/jpeg;base64,test-image-data-1' } } as any);
      }

      mockFileReader.result = 'data:image/png;base64/test-image-data-2';
      if (mockFileReader.onload) {
        mockFileReader.onload({ target: { result: 'data:image/png;base64/test-image-data-2' } } as any);
      }

      await waitFor(() => {
        expect(defaultProps.handleInputChange).toHaveBeenCalled();
      });
    }
  });

  it('handles image removal', () => {
    render(<ImagesSection {...defaultProps} />);

    const removeButtons = screen.getAllByRole('button');
    const removeButton = removeButtons.find((button) => button.textContent === '');

    if (removeButton) {
      fireEvent.click(removeButton);
      expect(defaultProps.handleInputChange).toHaveBeenCalledWith('imageURL', ['https://example.com/image2.jpg']);
    }
  });

  it('handles image load error', () => {
    render(<ImagesSection {...defaultProps} />);

    const image = screen.getByAltText('Room image 1');
    fireEvent.error(image);

    expect(image).toHaveStyle('display: none');
  });

  it('filters out null/undefined/empty images', () => {
    const roomWithMixedImages = {
      imageURL: ['https://example.com/image1.jpg', null, undefined, '', 'https://example.com/image2.jpg'],
    };

    render(<ImagesSection {...defaultProps} room={roomWithMixedImages} />);

    expect(screen.getByText('Image Previews (2)')).toBeInTheDocument();
    expect(screen.getByAltText('Room image 1')).toBeInTheDocument();
    expect(screen.getByAltText('Room image 2')).toBeInTheDocument();
  });

  it('displays correct image count', () => {
    const roomWithManyImages = {
      imageURL: Array.from({ length: 5 }, (_, i) => `https://example.com/image${i + 1}.jpg`),
    };

    render(<ImagesSection {...defaultProps} room={roomWithManyImages} />);

    expect(screen.getByText('Image Previews (5)')).toBeInTheDocument();
  });

  it('has correct file input attributes', () => {
    render(<ImagesSection {...defaultProps} />);

    const fileInput = document.querySelector('input[type="file"]');
    expect(fileInput).toHaveAttribute('multiple');
    expect(fileInput).toHaveAttribute('accept', 'image/*');
  });

  it('matches snapshot with images', () => {
    const { container } = render(<ImagesSection {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot without images', () => {
    const roomWithNoImages = {
      imageURL: [],
    };

    const { container } = render(<ImagesSection {...defaultProps} room={roomWithNoImages} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
