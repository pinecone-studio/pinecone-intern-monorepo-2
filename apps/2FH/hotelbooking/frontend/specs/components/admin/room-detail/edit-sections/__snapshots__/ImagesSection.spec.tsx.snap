// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesSection matches snapshot with images 1`] = `
<div
  class="space-y-6"
>
  <div>
    <label
      class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
    >
      Room Images
    </label>
    <div
      class="mt-2"
    >
      <button
        class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-background hover:bg-accent hover:text-accent-foreground px-4 py-2 w-full h-32 border-2 border-dashed border-gray-300 hover:border-gray-400 flex flex-col items-center justify-center gap-2"
        type="button"
      >
        <svg
          class="lucide lucide-upload text-gray-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
          />
          <polyline
            points="17 8 12 3 7 8"
          />
          <line
            x1="12"
            x2="12"
            y1="3"
            y2="15"
          />
        </svg>
        <span
          class="text-gray-600"
        >
          Click to upload images
        </span>
        <span
          class="text-sm text-gray-400"
        >
          PNG, JPG, GIF up to 10MB each
        </span>
      </button>
      <input
        accept="image/*"
        class="h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none disabled:cursor-not-allowed disabled:opacity-50 hidden"
        multiple=""
        type="file"
      />
    </div>
  </div>
  <div>
    <label
      class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
    >
      Image Previews (
      2
      )
    </label>
    <div
      class="mt-2 grid grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4"
    >
      <div
        class="relative group"
      >
        <div
          class="aspect-square rounded-lg overflow-hidden border border-gray-200"
        >
          <img
            alt="Room image 1"
            class="w-full h-full object-cover"
            src="https://example.com/image1.jpg"
          />
        </div>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground hover:bg-destructive/90 absolute -top-2 -right-2 h-6 w-6 rounded-full p-0 opacity-0 group-hover:opacity-100 transition-opacity"
          type="button"
        >
          <svg
            class="lucide lucide-x"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 6 6 18"
            />
            <path
              d="m6 6 12 12"
            />
          </svg>
        </button>
        <div
          class="absolute bottom-0 left-0 right-0 bg-black bg-opacity-50 text-white text-xs p-1 opacity-0 group-hover:opacity-100 transition-opacity"
        >
          Image 
          1
        </div>
      </div>
      <div
        class="relative group"
      >
        <div
          class="aspect-square rounded-lg overflow-hidden border border-gray-200"
        >
          <img
            alt="Room image 2"
            class="w-full h-full object-cover"
            src="https://example.com/image2.jpg"
          />
        </div>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground hover:bg-destructive/90 absolute -top-2 -right-2 h-6 w-6 rounded-full p-0 opacity-0 group-hover:opacity-100 transition-opacity"
          type="button"
        >
          <svg
            class="lucide lucide-x"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 6 6 18"
            />
            <path
              d="m6 6 12 12"
            />
          </svg>
        </button>
        <div
          class="absolute bottom-0 left-0 right-0 bg-black bg-opacity-50 text-white text-xs p-1 opacity-0 group-hover:opacity-100 transition-opacity"
        >
          Image 
          2
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ImagesSection matches snapshot without images 1`] = `
<div
  class="space-y-6"
>
  <div>
    <label
      class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
    >
      Room Images
    </label>
    <div
      class="mt-2"
    >
      <button
        class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-background hover:bg-accent hover:text-accent-foreground px-4 py-2 w-full h-32 border-2 border-dashed border-gray-300 hover:border-gray-400 flex flex-col items-center justify-center gap-2"
        type="button"
      >
        <svg
          class="lucide lucide-upload text-gray-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
          />
          <polyline
            points="17 8 12 3 7 8"
          />
          <line
            x1="12"
            x2="12"
            y1="3"
            y2="15"
          />
        </svg>
        <span
          class="text-gray-600"
        >
          Click to upload images
        </span>
        <span
          class="text-sm text-gray-400"
        >
          PNG, JPG, GIF up to 10MB each
        </span>
      </button>
      <input
        accept="image/*"
        class="h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none disabled:cursor-not-allowed disabled:opacity-50 hidden"
        multiple=""
        type="file"
      />
    </div>
  </div>
  <div
    class="text-center py-8 text-gray-500"
  >
    <svg
      class="lucide lucide-image mx-auto mb-2 text-gray-300"
      fill="none"
      height="48"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="18"
        rx="2"
        ry="2"
        width="18"
        x="3"
        y="3"
      />
      <circle
        cx="9"
        cy="9"
        r="2"
      />
      <path
        d="m21 15-3.086-3.086a2 2 0 0 0-2.828 0L6 21"
      />
    </svg>
    <p>
      No images uploaded yet
    </p>
    <p
      class="text-sm"
    >
      Upload images to preview them here
    </p>
  </div>
</div>
`;
